pkg/config/piped.go

```diff
@@ -258,21 +258,22 @@ type HelmChartRepositoryType string
 const (
 	HTTPHelmChartRepository HelmChartRepositoryType = "HTTP"
 	GITHelmChartRepository  HelmChartRepositoryType = "GIT"
+	OCIHelmChartRepository  HelmChartRepositoryType = "OCI"
 )
 
 type HelmChartRepository struct {
-	// The repository type. Currently, HTTP and GIT are supported.
+	// The repository type. Currently, HTTP, GIT and OCI are supported.
 	// Default is HTTP.
 	Type HelmChartRepositoryType `json:"type" default:"HTTP"`
 
 	// Configuration for HTTP type.
 	// The name of the Helm chart repository.
 	Name string `json:"name"`
-	// The address to the Helm chart repository.
+	// The address to the Helm chart repository or OCI registry.
```

Comment:
this confused me a bit. Is it okay to mention the OCI registry here? I mean, it looks like we have `chartRepositories` configuration, and under that section, we have a configuration to login to `xxxRegistry`, which is a bit weird to me (the registry is supposed to be the outer which contains many other repositories, I guess ). Maybe it's just the problem with what should we use to mention it in our codebase, but would like to hear other thoughts.

Reply:
I also would like to hear :bow: cc: @nghialv @knanao

Reply:
Sorry, I didn't notice this.
IMO, I think it's enough to be just only `The address to the Helm chart repository` because we can know what kind of the address by the `HelmChartRepositoryType`.
Besides, it's hard to explicit that one in all fields as the type increase.

Reply:
The following specifications may be clearer I think.
What do you think? (cc: @khanhtc1202)

1. Delete `OCI` type from `.spec.chartRepositories[_]` field in piped configuration.
2. Create `chartRegistries` field under the `.spec` field.
3. Create `address`, `username` and `password` fields under the `.spec.chartRegistries[_]` field.
4. Treat fields `.spec.chartRegistries[_].username` and `.spec.chartRegistries[_].password` as authentication credentials to login to specified registries in `.spec.chartRegistries[_].address`.

Reply:
👆 means we add a new configuration section named `chartRegistries` for the OCI registry, right? Then it's LGTM 👍  and I think we should add `type: OCI` to that section as well, wdyt?

Reply:
> 👆 means we add a new configuration section named chartRegistries for the OCI registry

Yes!

>  I think we should add type: OCI to that section as well, wdyt?

I would be fine either way, but I don't see any use for it other than the OCI Registry at of now, so I think it would be fine as future work.

Reply:
I get your point, but I still think the field like `type` should be added from the start, not as future work when needed, because later adding/changing field type would affect the running manifests. We can add that field and make the default value as `OCI` but should add that in the manifest struct definition.

Reply:
I understand. It certainly seemed to me, too, that it would be better to add it first for future use. 
I would like it to be in the same format as `type: OCI` as in `chartRepository`.

Reply:
@knanao @nghialv wdyt about the above suggestion 👀